modules/hal_nordic/nrfx/CMakeLists.txt

```
@@ -0,0 +1,40 @@
+# Copyright (c) 2025 Nordic Semiconductor ASA
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+zephyr_library()
+zephyr_library_property(ALLOW_EMPTY TRUE)
+
+# Inherit key properties from the upstream nrfx library.
+zephyr_library_get_current_dir_lib_name(${ZEPHYR_NRF_MODULE_DIR} zephyr_lib_nrfx)
+zephyr_library_include_directories($<TARGET_PROPERTY:${zephyr_lib_nrfx},INCLUDE_DIRECTORIES>)
+zephyr_library_compile_definitions($<TARGET_PROPERTY:${zephyr_lib_nrfx},COMPILE_DEFINITIONS>)
```

if i'm not mistaken, then all this code can be replaced by a single:
```suggestion
zephyr_library_amend()
```

---

Found a limitation upstream in `zephyr_library_amend()` which is fixed here: https://github.com/nrfconnect/sdk-zephyr/pull/2634

Including this Zephyr commit allow us to use: `zephyr_library_amend(${ZEPHYR_NRF_MODULE_DIR})`